homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Sensors for GoDice device."""
+
+import logging
+
+import godice
+
+from homeassistant import const
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DiceConfigEntry
+from .const import DICE_MAX_NUMBER, DICE_MIN_NUMBER, DOMAIN, MANUFACTURER, MODEL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ROLLED_NUMBER_SENSOR_DESCR = SensorEntityDescription(
+    key="rolled_number",
+    device_class=SensorDeviceClass.ENUM,
+    options=[str(num) for num in range(DICE_MIN_NUMBER, DICE_MAX_NUMBER + 1)],
+)
+COLOR_SENSOR_DESCR = SensorEntityDescription(
+    key="color",
+    device_class=SensorDeviceClass.ENUM,
+    options=[color.name.lower() for color in godice.Color],
+)
+BATTERY_SENSOR_DESCR = SensorEntityDescription(
+    key="battery_level",
+    device_class=SensorDeviceClass.BATTERY,
+    native_unit_of_measurement=const.PERCENTAGE,
+    state_class=SensorStateClass.MEASUREMENT,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: DiceConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Setups Dice sensors."""
+    dice = config_entry.runtime_data.dice
+    device_address = config_entry.data[const.CONF_ADDRESS]
```

Comment:
```suggestion
    device_address = config_entry.data[CONF_ADDRESS]
```
Please do this for the rest as well